internal/cri/server/podsandbox/controller.go

```diff
@@ -171,12 +171,16 @@ func (c *Controller) waitSandboxExit(ctx context.Context, p *types.PodSandbox, e
 		dctx, dcancel := context.WithTimeout(dctx, handleEventTimeout)
 		defer dcancel()
 		event := &eventtypes.TaskExit{ExitStatus: exitStatus, ExitedAt: protobuf.ToTimestamp(exitedAt)}
-		if cleanErr := handleSandboxTaskExit(dctx, p, event); cleanErr != nil {
+		if err := handleSandboxTaskExit(dctx, p, event); err != nil {
+			// TODO will backoff the event to the controller's own EventMonitor, but not cri's,
```

Comment:
Not sure I understand your comment correctly: maybe we should let CRI plugin to retry all the failures? so that we don't need to implement other retry for shim-type sandbox controller.

Reply:
`handleSandboxExit` in CRI plugin will only update the sandbox status in its `sandboxStore` and should ignore the implementation details of sandbox(So it should not know that if there is a `task` or `shim` for the sandbox), I think it is the podsandbox controller's responsibility to do cleanup of the legacy sandbox. and that is what we do in  `handleSandboxTaskExit`.

Reply:
And the spliting of EventMonitor of podsandbox from CRI plugin is done in #9598, The retry of sandbox container task cleanup is done in that PR